arelle/PackageManager.py

```diff
@@ -51,6 +51,27 @@ def langCloseness(l1, l2):
             return i
     return _len
 
+
+def _parseFile(cntlr, parser, filepath, file, schemaUrl):
+    """
+    Returns tree from `file`, parsed with `parser`, and validated against the provided schema at `schemaUrl`.
+    :return: Tree if parsed and validated, None if `schemaUrl` could not be loaded.
+    """
+    tree = etree.parse(file, parser=parser)
+    # schema validate tp xml
+    if cntlr.workingOnlineOrInCache(schemaUrl):
+        xsdTree = etree.parse(schemaUrl, parser=parser)
+        etree.XMLSchema(xsdTree).assertValid(tree)
+    else:
+        cntlr.addToLog(_("File could not be validated against the schema (%(schemaUrl)s) because the schema was not "
+                         "found in the cache and Arelle is configured to work offline."),
+                       messageArgs={"schemaUrl": schemaUrl},
+                       messageCode="tpe:workingOffline",
+                       file=filepath,
+                       level=logging.WARNING)
```

Comment:
I wonder if this should be an ERROR?  If you're offline without a taxonomy-package.xsd, then we can't know whether the taxonomy package is valid or not.

Reply:
The unknown aspect of that is why I opted for WARNING, we don't know there's an error. I don't know enough about the UX to say for sure, but my thinking was that at this point in the UX, the user isn't looking to validate the package, just load it into Arelle. I interpreted the validation here as a courtesy to give a "heads-up" what's been provided doesn't look like a valid taxonomy package, but I suppose the fact that validation errors [cause package manager to abort loading](https://github.com/Arelle/Arelle/blob/8a882f2fb2047d7bdd5e9249746e980b3085b5bb/arelle/PackageManager.py#L86) doesn't really align with that. I'm fine with bumping up the log level, but then I think the bigger question becomes: do we prevent the package from loading if we can't validate it?

Reply:
> at this point in the UX, the user isn't looking to validate the package

If that's true, then the schema validation is misplaced, and it should be wherever validations are found, but the presence of other `errors`/`cntlr.addToLog(level=ERROR)` just below make me think that this is the place in Arelle where invalid taxonomy package structures are caught.

I don't think it needs to be a fatal error that prevents opening, but it seems like it should be an ERROR level.